File: ruby/easy/baseball_game.rb
# @param {String[]} ops
# @return {Integer}
def cal_points(ops)
    stack = Array.new
    sum = 0
    ops.each do |op|
        if(op == "C")
            stack.pop
        elsif(op == "D")
            stack.push(stack.last * 2)
        elsif(op == "+")
            stack.push(stack.last + stack[stack.length-2])
        else
            stack.push(op.to_i)
        end
    end
    
    while !stack.empty?
        sum += stack.pop
    end
    return sum
end

File: javascript/easy/baseballGame.js
/**
 * @param {string[]} ops
 * @return {number}
 */
var calPoints = function(ops) {
    const stack = new Array();
    let sum = 0;
    for(let op of ops) {
        if(op === "C") {
            stack.pop();
        } else if(op === "D") {
            stack.push(stack[stack.length-1] * 2);
        } else if(op === "+") {
            stack.push(stack[stack.length-1] + stack[stack.length-2]);
        } else {
            stack.push(parseInt(op));
        }
    }
    
    while(stack.length > 0) {
        sum += stack.pop();
    }
    return sum;
};
